src/Persistent/Implementation.hs

```diff
@@ -0,0 +1,17 @@
+{-# LANGUAGE CPP #-}
```

Comment:
Oh, you don't only need to replace `mkRunPersist`, but also `Persist`.  Hum...  type families?  We may want to keep the CPP solution for now if that gets too complicated.